### PR TITLE
feat(components): tabs component add tabPosition prop

### LINKED ISSUE #2386: [Feature Request] Tab Vertical Orientation
### Is your feature request related to a problem? Please describe.

Would be great if we could have a vertical orientation for the Tabs  component.

### Describe the solution you'd like

I would like to be able to set an orientation={'vertical'} property on the Tab component in order to denote that it needs to go in portrait orientation down the page.

### Describe alternatives you've considered

I have built standard vertical menus using custom data properties and styling to indicate the selected tab, but having it in the concise Tab component would be very nice.

### Screenshots or Videos

_No response_

### PR DESCRIPTION
**Pull request by winchesHe**
Closes #2386 

## 📝 Description

> Tabs component add tabPosition prop

## 🚀 New behavior

> Add the position of the tabs

## 📝 Additional Information

Add Tabs Component props

1、placement

image

image

2、isVertical

image

image

## Summary by CodeRabbit

- **New Features**
	- Introduced customizable tab positions (top, bottom, start, end) with the new `tabPosition` property.
	- Added vertical orientation support with the `isVertical` property for both horizontal and vertical tab layouts.
	- Enhanced tab components to display photos, music, and videos content in various orientations.
- **Enhancements**
	- Implemented new styles for tab positions and orientations, including `tabList`, `panel`, and a new `wrapper` element.
- **Documentation**
	- Updated documentation to explain the usage of `tabPosition` and `isVertical` properties for personalized tab layouts.
- **Tests**
	- Included tests for validating tab positioning and orientation functionalities.

### CODE REVIEW

**Comment by kircsa08** [comment]
Is there planned approving date about this issue ? 

Thank you for replying!

**Comment by jrgarciadev** [comment]
Hey @winchesHe nice addition, could you please make these changes?

1. Rename the `left` and `right` values to `start` and `end`
2. Add a11y support for positioning 
3. Add a prop called `isVertical` to make the tab vertical (same as `start` / `end`)

Thanks! 🙏🏻

**Comment by winchesHe** [comment]
> Hey @winchesHe nice addition, could you please make these changes?
> 
> 1. Rename the `left` and `right` values to `start` and `end`
> 2. Add a11y support for positioning
> 3. Add a prop called `isVertical` to make the tab vertical (same as `start` / `end`)
> 
> Thanks! 🙏🏻

Got it, I will made these changes 👀

File: packages/core/theme/src/components/tabs.ts [hunk 1]
```diff
@@ -159,6 +160,22 @@ const tabs = tv({
...
+      },
+      end: {
+        tabList: "flex-col",
+        panel: "py-0 px-3",
+        wrapper: "flex flex-row-reverse",
+      },
+      bottom: {
+        wrapper: "flex flex-col-reverse",
+      },
+    },
```
Comment (jrgarciadev) [inline packages/core/theme/src/components/tabs.ts:178]: @winchesHe could you please check the RTL support?

Comment (winchesHe) [reply packages/core/theme/src/components/tabs.ts:178]: ok, I will check it

Comment (winchesHe) [reply packages/core/theme/src/components/tabs.ts:178]: image

image

I checked and it supports RTL very well

Comment (Innei) [reply packages/core/theme/src/components/tabs.ts:178]: space between the text and radio maybe had some style issue in RTL

**Review by jrgarciadev — APPROVED** [review]

**Comment by jrgarciadev** [comment]
@winchesHe could you please fix the conflicts?

**Comment by winchesHe** [comment]
> @winchesHe could you please fix the conflicts?

Quite sure

**Comment by jrgarciadev** [comment]
@winchesHe sorry but what do you think about changing the property name from `tabPosition` to `orientation` it better aligns with other components https://nextui.org/docs/components/radio-group#horizontal https://nextui.org/docs/components/checkbox-group#horizontal

File: apps/docs/content/components/tabs/placement.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+export default function App() {
+  const [position, setPosition] = React.useState("top");
+  return (
+    <div className="flex flex-col px-4">
+      <RadioGroup
+        className="mb-4"
+        value={position}
+        label="Placement"
+        orientation="horizontal"
```
Comment (jrgarciadev) [inline apps/docs/content/components/tabs/placement.ts:11]:
```suggestion
        orientation="top"
```

File: apps/docs/content/components/tabs/placement.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+const App = `import {Tabs, Tab, Card, CardBody, RadioGroup, Radio} from "@nextui-org/react";
+
+export default function App() {
+  const [position, setPosition] = React.useState("top");
```
Comment (jrgarciadev) [inline apps/docs/content/components/tabs/placement.ts:4]:
```suggestion
  const [placmenet, setPlacement] = React.useState("top");
```

File: apps/docs/content/components/tabs/placement.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        orientation="horizontal"
+        onValueChange={(value) => setPosition(value)}
+      >
+        <Radio value="top">top</Radio>
+        <Radio value="bottom">bottom</Radio>
+        <Radio value="start">start</Radio>
+        <Radio value="end">end</Radio>
+      </RadioGroup>
+      <div className="flex w-full flex-col">
+        <Tabs aria-label="Options" placement={position}>
```
Comment (jrgarciadev) [inline apps/docs/content/components/tabs/placement.ts:20]:
```suggestion
        <Tabs aria-label="Options" placement={placement}>
```

**Review by jrgarciadev — CHANGES_REQUESTED** [review]

**Pull request merged**